"Abacus is stuk" when using political votes group cache

When Abacus tries to use the cache for the CandidatesVotesForm, that fails with the console error `values.candidate_votes is undefined` and the user is shown the "Abacus is stuk"-page.

### Steps to reproduce
1. Fill in a polling station up to the first political group page (`/elections/1/data-entry/4/1/list/1`)
2. Fill in a number in any of the fields.
3. Navigate to any of the previous pages.
4. Navigate back to the political group page.
5. The "Abacus is stuk"-page is shown.

Note that at step 3, there's already a console error: `Uncaught TypeError: values.candidate_votes is undefined`:

![Image](https://github.com/user-attachments/assets/e000bfb0-da91-4fb9-adcf-1b422c3c6320)

The navigation does succeed. It's only when returning to the political group page, that the "Abacus is stuk"-page is shown.

### Analysis
`useCandidateVotes.ts` provides `cache.data as CandidateVotesValues` to `valuesToFormValues`. `cache.data` however is an array of `CandidateVotesValues`.

We don't have code coverage for the [relevant line in useCandidateVotes.ts](https://app.codecov.io/gh/kiesraad/abacus/blob/main/frontend%2Fsrc%2Ffeatures%2Fdata_entry%2Fcomponents%2Fcandidates_votes%2FuseCandidateVotes.ts).

> `useCandidateVotes.ts` provides `cache.data as CandidateVotesValues` to `valuesToFormValues`. `cache.data` however is an array of `CandidateVotesValues`.

This shows that using `as` in production code is a bad idea. Earlier this month I created an issue (#1301) to enable the `no-unsafe-type-assertion` ESLint rule. That rule catches this issue:

```
abacus/frontend/src/features/data_entry/components/candidates_votes/useCandidateVotes.ts
  17:30  error  Unsafe type assertion: type 'PoliticalGroupVotes' is more narrow than the original type  @typescript-eslint/no-unsafe-type-assertion
  19:13  error  Unsafe type assertion: type 'PoliticalGroupVotes' is more narrow than the original type  @typescript-eslint/no-unsafe-type-assertion
```

--- a/frontend/src/features/data_entry/components/candidates_votes/candidatesVotesValues.ts
+++ b/frontend/src/features/data_entry/components/candidates_votes/candidatesVotesValues.ts
@@ -8,7 +8,7 @@ export interface CandidateVotesFormValues extends Omit<PoliticalGroupVotes, "tot
 
 export type CandidateVotesValues = PoliticalGroupVotes;
 
-export function valuesToFormValues(values: PoliticalGroupVotes): CandidateVotesFormValues {
+export function valuesToFormValues(values: CandidateVotesValues): CandidateVotesFormValues {
   return {
     number: values.number,
     total: formatNumber(values.total),
@@ -16,7 +16,7 @@ export function valuesToFormValues(values: PoliticalGroupVotes): CandidateVotesF
   };
 }
 
-export function formValuesToValues(formData: CandidateVotesFormValues): PoliticalGroupVotes {
+export function formValuesToValues(formData: CandidateVotesFormValues): CandidateVotesValues {
   return {
     number: formData.number,
     total: deformatNumber(formData.total),

--- a/frontend/src/features/data_entry/components/candidates_votes/useCandidateVotes.ts
+++ b/frontend/src/features/data_entry/components/candidates_votes/useCandidateVotes.ts
@@ -12,12 +12,19 @@ import {
 export function useCandidateVotes(political_group_number: number) {
   const { onSubmit: _onSubmit, ...section } = useDataEntryFormSection<CandidateVotesFormValues>({
     section: `political_group_votes_${political_group_number}`,
-    getDefaultFormValues: (results, cache) =>
-      cache?.key === `political_group_votes_${political_group_number}`
-        ? valuesToFormValues(cache.data as CandidateVotesValues)
-        : valuesToFormValues(
-            results.political_group_votes.find((pg) => pg.number === political_group_number) as CandidateVotesValues,
-          ),
+    getDefaultFormValues: (results, cache) => {
+      let values;
+      if (cache?.key === `political_group_votes_${political_group_number}`) {
+        values = cache.data["political_group_votes"]?.find((pg) => pg.number === political_group_number);
+      }
+      values ??= results.political_group_votes.find((pg) => pg.number === political_group_number);
+      values ??= {
+        number: political_group_number,
+        total: 0,
+        candidate_votes: [],
+      };
+      return valuesToFormValues(values);
+    },
   });
 
   const [missingTotalError, setMissingTotalError] = React.useState(false);

